drivers/seesaw_soil/seesaw_soil.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2020 Viktor Gal
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @ingroup     drivers_seesaw_soil
+ * @{
+ *
+ * @file
+ * @brief       Driver for the Adafruit Seesaw Soil Moisture and Temperature Sensor.
+ *
+ * @author      Viktor Gal <viktor.gal@maeth.com>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "assert.h"
+#include "byteorder.h"
+#include "xtimer.h"
+#include "periph/i2c.h"
+#include "seesaw_soil.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+static int _seesaw_read_regs(const seesaw_soil_t *dev, uint16_t reg_addr, void* data, size_t len) {
+    int r;
+
+    reg_addr = htons(reg_addr);
+    // send the 16bit register address we want to read with STOP bit
```

Comment:
You should use C-style comment (here and below in this function):
```suggestion
    /* send the 16bit register address we want to read with STOP bit */
```